### ISSUE TITLE
Autodoc ignores documented instance/class attributes and comment when using typing definition

### ISSUE THREAD

**Issue by brunolaporais** [sphinx-doc/sphinx#4812]
Subject: autodoc ignores documented instance attributes when using typing definition.

### Problem
If typing is used in a instance attribute, all instance attributes in the class are ignored.

#### Procedure to reproduce the problem
```
class ExampleClass(object):
    """ Class docs """

    class_attribute_without_typing = None
    """This is a class attribute without typing. This comment is not ignored."""

    class_attribute_with_typing: str = None
    """This is a class attribute with typing. This comment is ignored."""

    def __init__(self):

        self.instance_attribute_without_typing = ""
        """This is an instance attribute without typing. This comment is not ignored."""

        self.instance_attribute_with_typing: str = ""
        """This is an instance attribute with typing. This comment is ignored. This attribute is also ignored."""
```

#### Error logs / results
Sphinx does not generate any warnings or errors.
Here's an example of unexpected documentation:
```
class ExampleClass[source]
    Bases: object

    Class docs
    class_attribute_with_typing = None

    class_attribute_without_typing = None
        This is a class attribute without typing. This comment is not ignored.

    instance_attribute_without_typing = None
        This is an instance attribute without typing. This comment is not ignored.
```

#### Expected results
Module attributes, class attributes, and instance attributes should be documented if they contain docstrings, even with typing definition.
```
class ExampleClass[source]
    Bases: object

    Class docs
    class_attribute_with_typing (str) = None
        This is a class attribute with typing. This comment is ignored.

    class_attribute_without_typing = None
        This is a class attribute without typing. This comment is not ignored.

    instance_attribute_without_typing = None
        This is an instance attribute without typing. This comment is not ignored.
    
    instance_attribute_with_typing (str) = None
        This is an instance attribute with typing. This comment is ignored. This attribute is also ignored.
```
### Environment info
- OS: Ubuntu 16.04, 64-bit
- Python version: 3.6.2
- Sphinx version: 1.8.0

**Comment by tk0miya** [comment]
Reproduced. Thank you for letting me know!

**Comment by tk0miya** [comment]
Fixed by #4813.
Thank you for reporting!
